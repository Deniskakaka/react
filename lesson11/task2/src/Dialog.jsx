import React from 'react';
import PropTypes from 'prop-types';

const Dialog = ({children, title, isOpen, close}) => {
    if (!isOpen) {
        return null;
    }
    return(
        <div className="dialog">
            <div className="dialog__heading">
                <div className="dialog__title">{title}</div>
                <button className="dialog__close-btn" onClick={close}>+</button>
            </div>
            <div className="dialog__content">{children}</div>
        </div>
    )
};

Dialog.propTypes = {
    isOpen: PropTypes.bool,
    children: PropTypes.element.isRequired,
    title: PropTypes.string,
    onClick: PropTypes.func.isRequired
}

Dialog.defaultProps = {
    title: '',
    isOpen: false
}

export default Dialog;
