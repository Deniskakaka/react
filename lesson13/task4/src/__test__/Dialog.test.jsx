import React from 'react';
import { shallow } from 'enzyme';
import Dialog from '../Dialog.jsx';

describe('Dialog', () => {
    it('should isOpen equal true dialog is show', () => {
        const wrappedComponent = shallow(<Dialog isOpen/>)
        expect(wrappedComponent.find('.dialog').exists()).toBeTruthy();
    });

    it('should isOpen equal false dialog is none', () => {
        const wrappedComponent = shallow(<Dialog />)
        expect(wrappedComponent.find('.dialog').exists()).toEqual(false);
    });

    it('should onClose is function and it work', () => {
        const mockLogin = jest.fn();
        const wrappedComponent = shallow(<Dialog isOpen onClose={mockLogin}/>)
        wrappedComponent.find('.dialog__close-btn').simulate('click')
        expect(mockLogin).toBeCalled();
    });

    it('should show display elements', () => {
        const wrappedComponent = shallow(<Dialog isOpen/>)
        expect(wrappedComponent).toMatchSnapshot();
    });

    it('should isOpen equal false dialog is none', () => {
        const wrappedComponent = shallow(<Dialog isOpen><span>0</span></Dialog>)
        expect(wrappedComponent.find('.dialog').find('.dialog__content').text()).toEqual('0');
    });
})