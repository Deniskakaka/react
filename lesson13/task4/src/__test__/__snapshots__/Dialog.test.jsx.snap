// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog should isOpen equal false dialog is none 1`] = `
<div
  className="dialog"
>
  <div
    className="dialog__heading"
  >
    <h4
      className="dialog__title"
    />
    <button
      className="dialog__close-btn"
    >
      +
    </button>
  </div>
  <div
    className="dialog__content"
  />
</div>
`;

exports[`Dialog should show display element 1`] = `
<div
  className="dialog__heading"
>
  <h4
    className="dialog__title"
  />
  <button
    className="dialog__close-btn"
  >
    +
  </button>
</div>
`;

exports[`Dialog should show display elements 1`] = `
<div
  className="dialog"
>
  <div
    className="dialog__heading"
  >
    <h4
      className="dialog__title"
    />
    <button
      className="dialog__close-btn"
    >
      +
    </button>
  </div>
  <div
    className="dialog__content"
  />
</div>
`;

exports[`Dialog should show display elements 2`] = `
<div
  className="dialog__heading"
>
  <h4
    className="dialog__title"
  />
  <button
    className="dialog__close-btn"
  >
    +
  </button>
</div>
`;
